app/js/services/DiseaseService.js

```diff
@@ -1,92 +1,109 @@
-function DiseaseService($q, $resource, AppSettings, DiseaseModel, $log, filterFilter, _, $localStorage, NotificationService) {
-  'ngInject';
-
-  $log = $log.getInstance('DiseaseService', false);
-  $log.log('');
-
-  const DISEASE_ENDPOINT  = `${AppSettings.api.baseUrl}${AppSettings.api.diseases}/`;
-  const DISEASES_RESOURCE = $resource(DISEASE_ENDPOINT,{},{ query:  {isArray:false} });
-  const service           = {};
- 
+function DiseaseService(
+  $q,
+  $resource,
+  AppSettings,
+  DiseaseModel,
+  $log,
+  filterFilter,
+  _,
+  $localStorage,
+  NotificationService
+) {
+  "ngInject";
+
+  $log = $log.getInstance("DiseaseService", false);
+  $log.log("");
+
+  const DISEASE_ENDPOINT = `${AppSettings.api.baseUrl}${
+    AppSettings.api.diseases
+  }/`;
+  const DISEASES_RESOURCE = $resource(
+    DISEASE_ENDPOINT,
+    {},
+    { query: { isArray: false } }
+  );
+  const service = {};
 
   // converts raw server response to array Disease Model promises
   // all models will be populated when resolved
-  let _responseTransformer = (serverResponse, mutationsGenes)=>{
-    return serverResponse.map((diseaseResponse)=>new DiseaseModel(diseaseResponse, mutationsGenes));
-  }
-
+  let _responseTransformer = serverResponse => {
+    return serverResponse.map(
+      diseaseResponse => new DiseaseModel(diseaseResponse)
+    );
+  };
 
   /**
    * Gets all disease resources either from the server or a local storage method
-   * converts them to DiseaseModels which come back as promises in order to allow them to fetch aggregate data
+   * converts them to DiseaseModels
    *
    * @param { String } searchQuery  -  query string passed from user input in QueryBuilder::onInputChange
    * @param { Object } dataSource  -   where to retrieve results from
    * @param { Array } mutationsGenes  -  array of gene entrezid, usually from queryBuilder user selected genes
    *
    * @return {Array} filtered array of DiseaseModels
    */
-  service.query = (searchQuery, mutationsGenes)=>{
-    
-    
-      $log.log(`query:${AppSettings.api.baseUrl}${AppSettings.api.diseases}/`);
-
-
-      let diseasePromise = new Promise((resolve, reject)=>{
-        if ($localStorage.diseaseData && $localStorage.diseaseData.count) {
-          if($localStorage.diseaseData.count === 0) return  reject(`No Disease Types found matching: "${searchQuery}"`);
-          $log.log('fetch disease data from localStorage',$localStorage.diseaseData);
-           resolve($localStorage.diseaseData); 
-        }
-
-        $log.log('fetch disease data from DB resource');
-
-        DISEASES_RESOURCE.query((diseaseResponse) => {
+  service.query = (searchQuery, mutationsGenes) => {
+    $log.log(`query:${AppSettings.api.baseUrl}${AppSettings.api.diseases}/`);
+
+    let diseasePromise = $q((resolve, reject) => {
```

Comment:
Hey, cool! I didn't realize `$q` could be called this way. I guess that made the API mapping even easier. 🥇